docs/src/components/ComponentDoc/ComponentExample/ComponentExample.js

```diff
@@ -64,36 +61,43 @@ class ComponentExample extends PureComponent {
     renderHtml: true,
   }
 
-  componentWillMount() {
-    const { examplePath } = this.props
-    this.anchorName = examplePathToHash(examplePath)
-
-    this.setState({
-      showCode: this.isActiveHash(),
-      sourceCode: this.getOriginalSourceCode(),
-    })
-  }
-
-  componentWillReceiveProps(nextProps) {
-    const { examplePath, exampleSources, location } = nextProps
-    const nextSourceCode = exampleSources[examplePath]
+  static getDerivedStateFromProps(props, state = {}) {
+    const {
+      examplePath,
+      exampleSources,
+      location: { hash },
+    } = props
+
+    const anchorName = examplePathToHash(examplePath)
+    const isActiveHash = anchorName === hash
```

Comment:
```suggestion
    const isActiveHash = anchorName === location.hash
```

Please use `location.hash`, without it's unclear what is it. It also will save us from nested destruction

Reply:
Done